src/pocketmine/item/Item.php

```diff
@@ -596,7 +596,13 @@ public function addEnchantment(Enchantment $ench){
 		}
 
 		if(!$found){
-			$tag->ench->{count($tag->ench) + 1} = new CompoundTag("", [
+			$count = 0;
+			foreach($tag->ench as $key => $value){
+				if(is_numeric($key)){
+					++$count;
+				}
+			}
+			$tag->ench->{$count + 1} = new CompoundTag("", [
```

Comment:
Isn't it supposed to be

$tag->ench{count($tag->ench)} = ...

rather than

$tag->ench{count($tag->ench) + 1} = ...


Count, unlike keys, doesn't take the first value of the array to be the 0th value.